packages/nimble-components/src/table/specs/data-hierarchy-hld.md

```diff
@@ -190,11 +199,22 @@ By making the `TableRecord` support hierarchy in its structure, it seemed possib
 -   There is no clear way to provide a strong type for `TableRecord` that would have a reserved field name of something like `subRows` that itself would be typed to an array of `TableRecord`.
 -   The performance profile between the prototypes of a hierarchical data structure, and a flat list were pretty close with one another.
 
+### Specify whether or not a row has delayed loaded children through the data records
+
+Each record passed to the table could have a boolean field that indicates whether or not it has delayed loaded data. The table then could have an attribute named `expansion-toggle-visible-field-name` that specifies the name of that field. The downside to this approach, however, is that it doesn't extend well to showing a loading indicator on the row. Specifically, we want to be able to toggle the loading state of a row without having to call `setData` on the table. Therefore, this approach would lead to two different APIs for specifying whether a row has delayed children and whether those children are currently being loaded.
+
+### Create a generic `setRowOptions` API that includes the hierarchy state
+
+The table could have a generic `setRowOptions` function instead of `setRecordHierarchyOptions`. The options set through that function could include delayed hierarchy information but also be extended easily in the future to include additional options. `setRecordHierarchyOptions` was chosen over a more generic `setRowOptions` function for the following reasons:
+
+-   Hierarchy options should be cleared when `parentIdFieldName` changes, but it doesn't make sense to clear non-hierarchical options in that case.
```

Comment:
```suggestion
-   Hierarchy options should be ignored when `parentIdFieldName` changes, but it doesn't make sense to clear non-hierarchical options in that case.
```

ignored right? we aren't clearing that cached state

Reply:
Good catch. This bullet point can be removed now since it is no longer a difference between how `setRecordHierarchyOptions` behaves and how we expect a future `setRowOptions` function to behave.